cloud/amazon/ec2_vpc_nat_gateway.py

```diff
@@ -0,0 +1,347 @@
+                    create_time = present_status.get('created')
+                    if present_status.get('state') == 'failed' and create_time.replace(tzinfo=None) >= start_time:
+                        module.fail_json(msg="Failed to create VPC NAT Gateway")
+                    elif present_status.get('state') == 'available':
+                        results['nat_gateway'] = camel_dict_to_snake_dict(result['NatGateway'])
+                        break
+                    else:
+                        time.sleep(5)
+            else:
+                module.fail_json(msg="Waited too long for VPC NAT Gateway to be created.")
+        module.exit_json(**results)
+    except botocore.exceptions.ClientError as ce:
+        module.fail_json(msg=ce.message, **camel_dict_to_snake_dict(ce.response))
+
+
+def ensure_nat_gateway_absent(ec2_client=None, module=None):
+    """ Ensure the specified NAT gateway does not exist and call delete if it does """
+    try:
+        results = dict(changed=False)
+        nat_gateways_result = ec2_client.describe_nat_gateways(NatGatewayIds=[module.params.get('nat_gateway_id')])
+        if nat_gateways_result:
+            nat_gateway = nat_gateways_result.get('NatGateways')[0]
+            if nat_gateway.get('State') in ('pending',
+                                            'available'):
+                if module.check_mode:
+                    results['changed'] = True
+                    module.exit_json(**results)
+                else:
+                    delete_nat_gateway(ec2_client=ec2_client,
+                                       module=module,
+                                       nat_gateway=nat_gateway)
+        module.exit_json(**results)
+    except botocore.exceptions.ClientError as ce:
+        module.fail_json(msg=ce.message, **camel_dict_to_snake_dict(ce.response))
+
+
+def ensure_nat_gateway_present(ec2_client=None, module=None):
+    """ Ensure NAT gateway with specified parameters exists and call create if it does not """
+    # Check allocation-id refers to existing EIP allocation
+    allocation_id = module.params.get('allocation_id')
+    try:
+        ec2_client.describe_addresses(AllocationIds=[allocation_id])
+
+        # Check subnet exists
+        subnet_id = module.params.get('subnet_id')
+        ec2_client.describe_subnets(SubnetIds=[subnet_id])
+
+        # Get existing number of NAT Gateways in specified subnet
+        nat_gateways = ec2_client.describe_nat_gateways(Filters=[{'Name': 'subnet-id',
+                                                                  'Values': [subnet_id]},
+                                                                 {'Name': 'state',
+                                                                  'Values': ['available', 'pending',
+                                                                             'deleting']}])
+        num_nat_gateways = len(nat_gateways.get('NatGateways')) if nat_gateways.get('NatGateways') else 0
```

Comment:
Can you use `len(nat_gateways.get('NatGateways', []))` instead to avoid the inline `if`?